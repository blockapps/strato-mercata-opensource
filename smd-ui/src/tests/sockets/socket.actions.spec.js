import {
  subscribeRoom,
  unSubscribeRoom
} from '../../sockets/socket.actions';
import {
  LAST_BLOCK_NUMBER,
  USERS_COUNT,
  CONTRACTS_COUNT,
  GET_TRANSACTIONS,
  BLOCKS_DIFFICULTY,
  BLOCKS_PROPAGATION,
  TRANSACTIONS_COUNT,
  TRANSACTIONS_TYPE,
  GET_PEERS,
  GET_COINBASE,
  BLOCKS_FREQUENCY,
  GET_NODE_UPTIME,
  GET_HEALTH,
  GET_SHARD_COUNT
} from '../../sockets/rooms';

describe('Socket: action', () => {

  describe('subscribe', () => {

    test('last block number', () => {
      expect(subscribeRoom(LAST_BLOCK_NUMBER)).toMatchSnapshot();
    });

    test('users count', () => {
      expect(subscribeRoom(USERS_COUNT)).toMatchSnapshot();
    });

    test('contracts count', () => {
      expect(subscribeRoom(CONTRACTS_COUNT)).toMatchSnapshot();
    });

    test('block propagation', () => {
      expect(subscribeRoom(BLOCKS_PROPAGATION)).toMatchSnapshot();
    });

    test('block frequency', () => {
      expect(subscribeRoom(BLOCKS_FREQUENCY)).toMatchSnapshot();
    });

    test('block difficulty', () => {
      expect(subscribeRoom(BLOCKS_DIFFICULTY)).toMatchSnapshot();
    });

    test('transactions count', () => {
      expect(subscribeRoom(TRANSACTIONS_COUNT)).toMatchSnapshot();
    });

    test('transactions type', () => {
      expect(subscribeRoom(TRANSACTIONS_TYPE)).toMatchSnapshot();
    });

    test('peers', () => {
      expect(subscribeRoom(GET_PEERS)).toMatchSnapshot();
    });

    test('coinbase', () => {
      expect(subscribeRoom(GET_COINBASE)).toMatchSnapshot();
    });

    test('transactions', () => {
      expect(subscribeRoom(GET_TRANSACTIONS)).toMatchSnapshot();
    });

    test('health', () => {
      expect(subscribeRoom(GET_HEALTH)).toMatchSnapshot();
    });

    test('uptime', () => {
      expect(subscribeRoom(GET_NODE_UPTIME)).toMatchSnapshot();
    });

    test('shard count', () => {
      expect(subscribeRoom(GET_SHARD_COUNT)).toMatchSnapshot();
    });
  })

  describe('unsubscribe', () => {

    test('last block number', () => {
      expect(unSubscribeRoom(LAST_BLOCK_NUMBER)).toMatchSnapshot();
    });

    test('users count', () => {
      expect(unSubscribeRoom(USERS_COUNT)).toMatchSnapshot();
    });

    test('contracts count', () => {
      expect(unSubscribeRoom(CONTRACTS_COUNT)).toMatchSnapshot();
    });

    test('block propagation', () => {
      expect(unSubscribeRoom(BLOCKS_PROPAGATION)).toMatchSnapshot();
    });

    test('block frequency', () => {
      expect(unSubscribeRoom(BLOCKS_FREQUENCY)).toMatchSnapshot();
    });

    test('blocks difficulty', () => {
      expect(unSubscribeRoom(BLOCKS_DIFFICULTY)).toMatchSnapshot();
    });

    test('transactions count', () => {
      expect(unSubscribeRoom(TRANSACTIONS_COUNT)).toMatchSnapshot();
    });

    test('transactions type', () => {
      expect(unSubscribeRoom(TRANSACTIONS_TYPE)).toMatchSnapshot();
    });

    test('coinbase', () => {
      expect(unSubscribeRoom(GET_COINBASE)).toMatchSnapshot();
    });

    test('peers', () => {
      expect(unSubscribeRoom(GET_PEERS)).toMatchSnapshot();
    });

    test('transactions', () => {
      expect(unSubscribeRoom(GET_TRANSACTIONS)).toMatchSnapshot();
    });

    test('health', () => {
      expect(unSubscribeRoom(GET_HEALTH)).toMatchSnapshot();
    });

    test('uptime', () => {
      expect(unSubscribeRoom(GET_NODE_UPTIME)).toMatchSnapshot();
    });

    test('shard count', () => {
      expect(unSubscribeRoom(GET_SHARD_COUNT)).toMatchSnapshot();
    });


  })

});
