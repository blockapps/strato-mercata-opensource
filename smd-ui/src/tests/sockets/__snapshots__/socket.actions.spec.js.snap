// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Socket: action subscribe block difficulty 1`] = `
Object {
  "name": "SUBSCRIBE/BLOCKS_DIFFICULTY",
  "type": "SOCKET_SUBSCRIBE_ROOM",
}
`;

exports[`Socket: action subscribe block frequency 1`] = `
Object {
  "name": "SUBSCRIBE/BLOCKS_FREQUENCY",
  "type": "SOCKET_SUBSCRIBE_ROOM",
}
`;

exports[`Socket: action subscribe block propagation 1`] = `
Object {
  "name": "SUBSCRIBE/BLOCKS_PROPAGATION",
  "type": "SOCKET_SUBSCRIBE_ROOM",
}
`;

exports[`Socket: action subscribe coinbase 1`] = `
Object {
  "name": "SUBSCRIBE/GET_COINBASE",
  "type": "SOCKET_SUBSCRIBE_ROOM",
}
`;

exports[`Socket: action subscribe contracts count 1`] = `
Object {
  "name": "SUBSCRIBE/CONTRACTS_COUNT",
  "type": "SOCKET_SUBSCRIBE_ROOM",
}
`;

exports[`Socket: action subscribe health 1`] = `
Object {
  "name": "SUBSCRIBE/GET_HEALTH",
  "type": "SOCKET_SUBSCRIBE_ROOM",
}
`;

exports[`Socket: action subscribe last block number 1`] = `
Object {
  "name": "SUBSCRIBE/LAST_BLOCK_NUMBER",
  "type": "SOCKET_SUBSCRIBE_ROOM",
}
`;

exports[`Socket: action subscribe peers 1`] = `
Object {
  "name": "SUBSCRIBE/GET_PEERS",
  "type": "SOCKET_SUBSCRIBE_ROOM",
}
`;

exports[`Socket: action subscribe shard count 1`] = `
Object {
  "name": "SUBSCRIBE/GET_SHARD_COUNT",
  "type": "SOCKET_SUBSCRIBE_ROOM",
}
`;

exports[`Socket: action subscribe transactions 1`] = `
Object {
  "name": "SUBSCRIBE/GET_TRANSACTIONS",
  "type": "SOCKET_SUBSCRIBE_ROOM",
}
`;

exports[`Socket: action subscribe transactions count 1`] = `
Object {
  "name": "SUBSCRIBE/TRANSACTIONS_COUNT",
  "type": "SOCKET_SUBSCRIBE_ROOM",
}
`;

exports[`Socket: action subscribe transactions type 1`] = `
Object {
  "name": "SUBSCRIBE/TRANSACTIONS_TYPE",
  "type": "SOCKET_SUBSCRIBE_ROOM",
}
`;

exports[`Socket: action subscribe uptime 1`] = `
Object {
  "name": "SUBSCRIBE/GET_NODE_UPTIME",
  "type": "SOCKET_SUBSCRIBE_ROOM",
}
`;

exports[`Socket: action subscribe users count 1`] = `
Object {
  "name": "SUBSCRIBE/USERS_COUNT",
  "type": "SOCKET_SUBSCRIBE_ROOM",
}
`;

exports[`Socket: action unsubscribe block frequency 1`] = `
Object {
  "name": "UNSUBSCRIBE/BLOCKS_FREQUENCY",
  "type": "SOCKET_UNSUBSCRIBE_ROOM",
}
`;

exports[`Socket: action unsubscribe block propagation 1`] = `
Object {
  "name": "UNSUBSCRIBE/BLOCKS_PROPAGATION",
  "type": "SOCKET_UNSUBSCRIBE_ROOM",
}
`;

exports[`Socket: action unsubscribe blocks difficulty 1`] = `
Object {
  "name": "UNSUBSCRIBE/BLOCKS_DIFFICULTY",
  "type": "SOCKET_UNSUBSCRIBE_ROOM",
}
`;

exports[`Socket: action unsubscribe coinbase 1`] = `
Object {
  "name": "UNSUBSCRIBE/GET_COINBASE",
  "type": "SOCKET_UNSUBSCRIBE_ROOM",
}
`;

exports[`Socket: action unsubscribe contracts count 1`] = `
Object {
  "name": "UNSUBSCRIBE/CONTRACTS_COUNT",
  "type": "SOCKET_UNSUBSCRIBE_ROOM",
}
`;

exports[`Socket: action unsubscribe health 1`] = `
Object {
  "name": "UNSUBSCRIBE/GET_HEALTH",
  "type": "SOCKET_UNSUBSCRIBE_ROOM",
}
`;

exports[`Socket: action unsubscribe last block number 1`] = `
Object {
  "name": "UNSUBSCRIBE/LAST_BLOCK_NUMBER",
  "type": "SOCKET_UNSUBSCRIBE_ROOM",
}
`;

exports[`Socket: action unsubscribe peers 1`] = `
Object {
  "name": "UNSUBSCRIBE/GET_PEERS",
  "type": "SOCKET_UNSUBSCRIBE_ROOM",
}
`;

exports[`Socket: action unsubscribe shard count 1`] = `
Object {
  "name": "UNSUBSCRIBE/GET_SHARD_COUNT",
  "type": "SOCKET_UNSUBSCRIBE_ROOM",
}
`;

exports[`Socket: action unsubscribe transactions 1`] = `
Object {
  "name": "UNSUBSCRIBE/GET_TRANSACTIONS",
  "type": "SOCKET_UNSUBSCRIBE_ROOM",
}
`;

exports[`Socket: action unsubscribe transactions count 1`] = `
Object {
  "name": "UNSUBSCRIBE/TRANSACTIONS_COUNT",
  "type": "SOCKET_UNSUBSCRIBE_ROOM",
}
`;

exports[`Socket: action unsubscribe transactions type 1`] = `
Object {
  "name": "UNSUBSCRIBE/TRANSACTIONS_TYPE",
  "type": "SOCKET_UNSUBSCRIBE_ROOM",
}
`;

exports[`Socket: action unsubscribe uptime 1`] = `
Object {
  "name": "UNSUBSCRIBE/GET_NODE_UPTIME",
  "type": "SOCKET_UNSUBSCRIBE_ROOM",
}
`;

exports[`Socket: action unsubscribe users count 1`] = `
Object {
  "name": "UNSUBSCRIBE/USERS_COUNT",
  "type": "SOCKET_UNSUBSCRIBE_ROOM",
}
`;
